### PR TITLE
WIP: Add e2e example with documentation

### PR DESCRIPTION
**Pull request by Qinusty**
This PR adds documentation for deploying the various sub-projects locally alongside package-feeds and also provides the necessary configs and docker-compose for doing such. This also improves the documentation for the environment variables used in the various sub projects throughout package-analysis.

NOTE: This example currently does not support the fillindexes or server binaries. This can be added at a later date.

### CODE REVIEW

**Comment by Qinusty** [comment]
@oliverchang After this merges we will need to run `build_docker.sh` with credentials to push the new images including scheduler. We should probably aim to integrate this into the CI so it happens on merge to master.

Also, with the various drivers to gocloud dev it might be useful to identify a subset of drivers which will be supported and imported throughout to avoid the need for PRs adding popular drivers when needed. An example of this is the buckets where S3 would be a popular addition to support AWS S3 and S3 compatible APIs.

**Comment by tom--pollard** [comment]
This covers https://github.com/ossf/package-analysis/issues/72

File: examples/e2e/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+services:
```
Comment (tom--pollard) [inline examples/e2e/docker-compose.yml:1]: I needed `version: "3"` at the start for my local `docker-compose` (1.25.0) to parse this

**Comment by tom--pollard** [comment]
After testing this locally (and jumping through a few hoops with my local docker install) I hit the following issue once it has published one set of package results

```
analysis_1   | 2021/06/03 15:10:48 uploading to bucket=file:///data/, path=pypi/grolt/1.0.0.json
analysis_1   | 2021/06/03 15:10:48 Failed to write to docstore = open docstore.Collection: no scheme in URL "files?name_field=ID"
analysis_1   | panic: Failed to write to docstore = open docstore.Collection: no scheme in URL "files?name_field=ID"
analysis_1   | 
analysis_1   | 
analysis_1   | goroutine 1 [running]:
analysis_1   | log.Panicf(0xe44e97, 0x21, 0xc0006b5e58, 0x1, 0x1)
analysis_1   | 	/usr/local/go/src/log/log.go:361 +0xc5
analysis_1   | main.messageLoop(0xf45020, 0xc000040020, 0xc0000cc000, 0xc00003c01e, 0xd, 0x0, 0x0)
analysis_1   | 	/src/cmd/worker.go:61 +0x745
analysis_1   | main.main()
analysis_1   | 	/src/cmd/worker.go:80 +0x114
e2e_analysis_1 exited with code 2
```

**Comment by oliverchang** [comment]
Thanks a lot for setting this up!

I uploaded https://github.com/ossf/package-analysis/pull/85/files to make both the bucket and docstore env vars optional, which should fix your issues.

File: README.md [hunk 2]
```diff
@@ -16,6 +16,21 @@ infrastructure to study behavior of open source packages and to look for malicio
...
+
+An example of these variables can be found in the [e2e example docker-compose](examples/e2e/docker-compose.yml).
+### Analysis
+
+`OSSMALWARE_WORKER_SUBSCRIPTION` - Can be used to set the subscription URL for the data coming out of scheduler. Values should follow [goclouddev subscriptions](https://gocloud.dev/howto/pubsub/subscribe/).
+`OSSF_MALWARE_ANALYSIS_RESULTS` - Can be used to set the bucket URL to publish results to. Values should follow [goclouddev buckets](https://gocloud.dev/howto/blob/).
```
Comment (oliverchang) [inline README.md:27]: Also `OSSMALWARE_DOCSTORE_URL`.

File: scheduler/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,11 @@
+FROM golang:1.16.3 as build
```
Comment (oliverchang) [inline scheduler/Dockerfile:1]: Can you pin this as well? see https://github.com/ossf/package-analysis/blob/main/analysis/Dockerfile#L1 for an example

File: scheduler/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+WORKDIR /src
+COPY . ./
+RUN go build -o scheduler main.go
+
+
+FROM ubuntu:21.04@sha256:be154cc2b1211a9f98f4d708f4266650c9129784d0485d4507d9b0fa05d928b6
```
Comment (oliverchang) [inline scheduler/Dockerfile:7]: Let's use something more lightweight here like distroless: e.g. https://github.com/ossf/scorecard/blob/d528b6e6262fbd8667bde1f7c8ad38078953cdb4/Dockerfile#L28

**Review by oliverchang — APPROVED** [review]

**Pull request merged**